llama_stack/distribution/server/routes.py

```diff
@@ -83,37 +81,44 @@ def _convert_path_to_regex(path: str) -> str:
 
         return f"^{pattern}$"
 
-    for api, api_endpoints in endpoints.items():
+    for api, api_routes in routes.items():
         if api not in impls:
             continue
-        for endpoint in api_endpoints:
+        for route in api_routes:
             impl = impls[api]
-            func = getattr(impl, endpoint.name)
-            if endpoint.method not in endpoint_impls:
-                endpoint_impls[endpoint.method] = {}
-            endpoint_impls[endpoint.method][_convert_path_to_regex(endpoint.route)] = (
+            func = getattr(impl, route.name)
+            # Get the first (and typically only) method from the set, filtering out HEAD
+            available_methods = [m for m in route.methods if m != "HEAD"]
+            if not available_methods:
+                continue  # Skip if only HEAD method is available
+            method = available_methods[0].lower()
+            if method not in endpoint_impls:
+                endpoint_impls[method] = {}
+            endpoint_impls[method][_convert_path_to_regex(route.path)] = (
                 func,
-                endpoint.descriptive_name or endpoint.route,
+                route.path,
             )
 
     return endpoint_impls
 
 
-def find_matching_endpoint(method, path, endpoint_impls):
+def find_matching_route(
+    method: str, path: str, route_impls: dict[str, dict[str, tuple[Callable[..., Any], str]]]
```

Comment:
omg that is some type frankenstein. can you add some type alias to make this a bit more parsable? like `PathImpl = dict[str, tuple[...]]` etc.